File: genAI/models/summaryModels.py
from pydantic import BaseModel


# Shared request model for analytics and insights
class ReceiptItem(BaseModel):
    name: str
    price: float
    quantity: int = 1
    category: str


class UserReceipt(BaseModel):
    merchant: str
    date: str
    totalAmount: float
    items: list[ReceiptItem]


class SummaryRequest(BaseModel):
    userID: str
    currency: str
    period: str
    receipts: list[UserReceipt]


# Analytics response models
class CategoryItem(BaseModel):
    category: str
    amount: float


class DailySpending(BaseModel):
    date: str
    amount: float


class GetAnalyticsResponse(BaseModel):
    totalAmount: float
    categoryBreakdown: list[CategoryItem]
    dailySpending: list[DailySpending]


# Insights response model
class GetInsightsResponse(BaseModel):
    summary: str
    warnings: list[str]

File: backend/models/summaryModels.go
package models

import "encoding/json"

// --- GenAI Request structs ---

type GenAISummaryReceiptItem struct {
	Name     string  `json:"name"`
	Price    float64 `json:"price"`
	Quantity int     `json:"quantity"`
	Category string  `json:"category"`
}

type GenAISummaryReceipt struct {
	Merchant    string                    `json:"merchant"`
	Date        string                    `json:"date"`
	TotalAmount float64                   `json:"totalAmount"`
	Items       []GenAISummaryReceiptItem `json:"items"`
}

type GenAISummaryRequest struct {
	UserID   string                `json:"userID"`
	Currency string                `json:"currency"`
	Period   string                `json:"period"`
	Receipts []GenAISummaryReceipt `json:"receipts"`
}

// --- GenAI Response structs ---

type CategoryBreakdownItem struct {
	Category string  `json:"category"`
	Amount   float64 `json:"amount"`
}

type DailySpendingItem struct {
	Date   string  `json:"date"`
	Amount float64 `json:"amount"`
}

type GenAIAnalyticsResponse struct {
	TotalAmount       float64                 `json:"totalAmount"`
	CategoryBreakdown []CategoryBreakdownItem `json:"categoryBreakdown"`
	DailySpending     []DailySpendingItem     `json:"dailySpending"`
}

type GenAIInsightsResponse struct {
	Summary  string   `json:"summary"`
	Warnings []string `json:"warnings"`
}

// --- Backend Response structs (served to frontend) ---

type CachedAnalyticsResponse struct {
	TotalSpent        float64         `json:"totalSpent"`
	Currency          string          `json:"currency"`
	CategoryBreakdown json.RawMessage `json:"categoryBreakdown"`
	DailySpending     json.RawMessage `json:"dailySpending"`
	Period            string          `json:"period"`
	ComputedAt        string          `json:"computedAt"`
}

type CachedInsightsResponse struct {
	Summary    string          `json:"summary"`
	Warnings   json.RawMessage `json:"warnings"`
	Period     string          `json:"period"`
	ComputedAt string          `json:"computedAt"`
}
